docs/tutorials/text_sentiment_ngrams.ipynb

```diff
@@ -0,0 +1,419 @@
+    "        super().__init__()\n",
+    "        self.embedding = nn.EmbeddingBag(vocab_size, embed_dim, sparse=True)\n",
+    "        self.fc = nn.Linear(embed_dim, num_class)\n",
+    "        self.init_weights()\n",
+    "\n",
+    "    def init_weights(self):\n",
+    "        initrange = 0.5\n",
+    "        self.embedding.weight.data.uniform_(-initrange, initrange)\n",
+    "        self.fc.weight.data.uniform_(-initrange, initrange)\n",
+    "        self.fc.bias.data.zero_()\n",
+    "        \n",
+    "    def forward(self, text, offsets):\n",
+    "        embedded = self.embedding(text, offsets)\n",
+    "        return self.fc(embedded)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initiate an instance\n",
+    "\n",
+    "The AG_NEWS dataset has four labels and therefore the number of classes is four.\n",
+    "\n",
+    "    1 : World\n",
+    "    2 : Sports\n",
+    "    3 : Business\n",
+    "    4 : Sci/Tec\n",
+    "\n",
+    "The vocab size is equal to the length of vocab (including single word and ngrams). The number of classes is equal to the number of labels, which is four in AG_NEWS case."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "VOCAB_SIZE = len(train_dataset.get_vocab())\n",
+    "EMBED_DIM = 32\n",
+    "NUN_CLASS = len(train_dataset.get_labels())\n",
+    "model = TextSentiment(VOCAB_SIZE, EMBED_DIM, NUN_CLASS).to(device)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Functions used to generate batch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since the text entries have different lengths, a custom function generate_batch() is used to generate data batches and offsets. The function is passed to `collate_fn` in `torch.utils.data.DataLoader`. The input to `collate_fn` is a list of tensors with the size of batch_size, and the `collate_fn` function packs them into a mini-batch. Pay attention here and make sure that `collate_fn` is declared as a top level def. This ensures that the function is available in each worker.\n",
+    "\n",
+    "The text entries in the original data batch input are packed into a list and concatenated as a single tensor as the input of `nn.EmbeddingBag`. The offsets is a tensor of delimiters to represent the beginning index of the individual sequence in the text tensor. Label is a tensor saving the labels of individual text entries."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def generate_batch(batch):\n",
+    "    label = torch.tensor([entry[0] for entry in batch])\n",
+    "    text = [entry[1] for entry in batch]\n",
+    "    offsets = [0] + [len(entry) for entry in text]\n",
+    "    offsets = torch.tensor(offsets[:-1]).cumsum(dim=0)\n",
```

Comment:
I think this call into cumsum can be a bit confusing. It could be good to spell out an example in the text above.